src/openforms/js/components/admin/form_design/logic/actions/Actions.js

```diff
@@ -109,6 +109,35 @@ const ActionVariableValue = ({action, errors, onChange}) => {
   );
 };
 
+const ActionFetchFromService = ({action, errors, onChange}) => {
+  const formContext = useContext(FormContext);
+  const allVariables = formContext.formVariables;
+  debugger;
+
+  return (
+    <>
+      <DSLEditorNode errors={errors.variable}>
+        <Select
+          name="variable"
+          choices={getVariableChoices(allVariables)}
+          allowBlank
+          onChange={onChange}
+          value={action.variable}
+        />
```

Comment:
Well I expect this "variable" choice wil not be needed any more as it should be the FormVariable.

Instead of pre-emptively changing stuff, I'd like to see what @stevenbal comes up with for the modal ui.

Reply:
I left the ActionVariable as is; the suggestion changes the semantics of what happens: the select now offers a superset the union of component variables and user defined variables. The ComponentSelection component offers just the components.